pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -115,6 +117,69 @@ def test_groupby_aggregation_mixed_dtype():
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_aggregation_non_numeric_dtype():
+    # GH #43108
+    df = DataFrame(
+        [["M", [1]], ["M", [1]], ["W", [10]], ["W", [20]]], columns=["MW", "v"]
+    )
+
+    expected = DataFrame(
+        {
+            "v": [[1, 1], [10, 20]],
+        },
+        index=Index(["M", "W"], dtype="object", name="MW"),
+    )
+
+    with tm.assert_produces_warning(UserWarning):
+        g = df.groupby(by=["MW"])
+        result = g.sum()
+        tm.assert_frame_equal(result, expected)
```

Comment:
try to put the minimum amount of code inside the tm.assert_produces_warning context.  I think just the .sum() call produces the warning, so the other two lines would go outside

Reply:
yeah I am making all the necessary changes